test/test.js

```diff
@@ -890,340 +919,443 @@ describe('S3rver Tests', function () {
   });
 
   it('should reach the server with a bucket vhost', function (done) {
-    request({ url: 'http://localhost:4569/', headers: { host: buckets[0] + '.s3.amazonaws.com' } }, function (err, response, body) {
-      if (err) {
-        return done(err);
-      }
+    request({ url: s3Client.endpoint.href, headers: { host: buckets[0] + '.s3.amazonaws.com' } }, function (err, response, body) {
+      if (err) return done(err);
 
-      if (response.statusCode !== 200) {
-        return done(new Error('Invalid status: ' + response.statusCode));
-      }
+      response.statusCode.should.equal(200);
+      body.should.containEql('ListBucketResult');
+      done();
+    });
+  });
+});
 
-      if (body.indexOf('ListBucketResult') === -1) {
-        return done(new Error('Unexpected response: ' + body));
-      }
+describe('S3rver CORS Policy Tests', function () {
+  const bucket = 'foobars';
+  let s3Client;
```

Comment:
Could also share `S3rver` as a dependency across tests so you can add an `after` hook for cleanup, and `afterEach` hook to close the server so that you don't need to do it inside each individual test.